include/libopencm3/cm3/memorymap.h

```diff
@@ -21,7 +21,11 @@
 #define LIBOPENCM3_CM3_MEMORYMAP_H
 
 #ifndef __CMSIS_USE
-#if defined(__CM3_CMSIS_VERSION_MAIN) || defined(__CM4_CMSIS_VERSION_MAIN)
+#if defined(__STM32F0xx_CMSIS_DEVICE_VERSION) \
+            || defined(__STM32F1xx_CMSIS_DEVICE_VERSION)\
+            || defined(__STM32F2xx_CMSIS_DEVICE_VERSION)\
+            || defined(__STM32F3xx_CMSIS_DEVICE_VERSION)\
+            || defined(__STM32F4xx_CMSIS_DEVICE_VERSION)
```

Comment:
hm. i thought, you wanted reject this idea anyway. 
it`s "even work" on that noted platforms... as for other targets  - somebody who need this compilance, should append their identificators. 
if you realy, interested in that patch, i append other identificators, that i can found in cmsis for stm32
this patch is not complete solution, cause i have compile it vs only rcc, gpio, and usb parths. other devices not tested vs macro definitions conflicts.